packages/test/test-end-to-end-tests/src/test/batching.spec.ts

```diff
@@ -344,6 +346,103 @@ describeFullCompat("Flushing ops", (getTestObjectProvider) => {
             });
         });
 
+        describe("TurnBased flushing of batches with compression", () => {
```

Comment:
You should also check out `packages/test/test-end-to-end-tests/src/test/messageSize.spec.ts`, write a test which enables compression and see it in action solving the 1MB problem 😁

Reply:
Added two tests in latest commit (one will fail until I handle the large single op case correctly in another commit)